bin/utest/signal.c

```diff
@@ -67,11 +68,14 @@ int test_signal_abort(void) {
 }
 
 /* ======= signal_segfault ======= */
-/* This test shall be considered success if the process gets terminated with
-   SIGABRT */
 int test_signal_segfault(void) {
   volatile struct { int x; } *ptr = 0x0;
-  ptr->x = 42;
+
+  siginfo_t si;
+  TEST_EXPECT_SIGNAL(SIGSEGV, &si, ptr, SEGV_MAPERR) {
```

Comment:
That looks much better but... since you pass extra arguments to `TEST_EXPECT_SIGNAL` it cannot be used to test the reception of other signals – think of @pwit81 who will continue work on `SIGILL`, `SIGTRAP`, `SIGBUS` and so on.

After the loop you could use another macro `GOT_SIGSEGV(&si, ptr, SEGV_MAPERR)`.

Or perhaps you could build `EXPECT_SIGSEGV` on top of `EXPECT_SIGNAL` ?